fpt/src/memory.rs

```diff
@@ -242,7 +242,7 @@ impl Memory {
         &mut self.mem[range]
     }
 
-    pub fn code_listing(&self) -> &[Option<String>; 0xffff + 1] {
+    pub fn code_listing(&self) -> &Vec<Option<String>> {
```

Comment:
btw the return type can also be `&[Option<String>]` because `Vec<T>` implements the `Deref` and `DerefMut` traits with `type Target = T` (which is `String` in our case)

Reply:
👍